source/sigv4_http_request.c

```diff
@@ -294,180 +294,6 @@ struct aws_signable *aws_signable_new_http_request(struct aws_allocator *allocat
     return NULL;
 }
 
-/*
- * Utility struct/API to let us wait on credentials resolution
- */
-struct aws_signing_waiter {
```

Comment:
Deleting this big chunk of code because it wasn't being used by anything

Reply:
It used to be used by elasticurl via dynamic load/bind, but that's a terrible solution and I wish I'd never done it.